### PR TITLE
Alfa Budget App

### PR DESCRIPTION
**Pull request by aliatayee**
## Alfa Budget App - Capstone Project
In this milestone I have built the final capstone project with the following requirments:

### [Demo Link](https://guarded-anchorage-90290.herokuapp.com/)

transactions_list
#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,81 @@
...
 
-* Ruby version
+## Major Technologies
+- [ ] Ruby on Rails
+- [ ] Postgresql
+- [ ] TailwindCSS
 
-* System dependencies
+### [Video Presentation Link](https://www.loom.com/share/027dc24ffb754aab86db457fddc06530)
+### Install & Setup
```
Comment (elisha2kyakpo1) [inline README.md]: - [ ] I did not see a piece of code that is doing expense calculations either in the video or in the project. Please you can point me in the right direction to that code.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @aliatayee, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The PR title and description looks professional✔️
- Descriptive commit messages✔️
- You have a descriptive readme file✔️
- No linter errors✔️

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly deploy the app and provide a link to the deployed app in the readME file as per project requirements.
- [ ] I have failed to test your app because I can't signup, the only button I see as a new user is the sign-in. 

signup


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 2]
```diff
@@ -1,24 +1,81 @@
...
 
-* Configuration
+To setup and install this sample AlFa Blog project, follow the below steps:
+- Clone this project by the command: 
+  ```
+  git clone https://github.com/aliatayee/alfa-budget-app
+  ```
```
Comment (rotimiazeez) [inline README.md:20]: - For your app deployment, I'd suggest you use [Railway](railway.app). 😉

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @aliatayee 👍🏽, 
## Great work on making the changes 👏 
### You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by rloterh — COMMENTED** [review]
Hi Ali,
 
 While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
 
 ## Invalid Code Review Request
You have submitted a project, whereby linter checks are failing on your Pull Request. Kindly either remove the unrequired linter or ensure that **all** checks are passing and request another code review.

**linter failing:**
image

 
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
 Cheers and Happy coding!👏👏👏
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _Invalid Code Review Request **does not count** into the code reviews limit._

**Review by medaminedev66 — APPROVED** [review]
Hi @aliatayee 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 💯 
Congratulations! 🎉 

### Highlights
- Your repo looks professional.
- The project is deployed properly, and your web app works as expected.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**